hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java

```diff
@@ -246,69 +247,105 @@ private long eventsProcessed() {
     public EventRegistration registerLocalListener(@Nonnull String serviceName,
                                                    @Nonnull String topic,
                                                    @Nonnull Object listener) {
-        return registerListenerInternal(serviceName, topic, TrueEventFilter.INSTANCE, listener, true);
+        return registerLocalListener(serviceName, topic, TrueEventFilter.INSTANCE, listener);
     }
 
     @Override
-    public EventRegistration registerLocalListener(@Nonnull String serviceName,
-                                                   @Nonnull String topic,
-                                                   @Nonnull EventFilter filter,
-                                                   @Nonnull Object listener) {
-        return registerListenerInternal(serviceName, topic, filter, listener, true);
+    public EventRegistration registerLocalListener(@Nonnull String serviceName, @Nonnull String topic,
+                                                   @Nonnull EventFilter filter, @Nonnull Object listener) {
+        return registerListenerInternalLocally(serviceName, topic, filter, listener, true);
+    }
+
+    /**
+     *
+     /**
+     * Registers the listener for events matching the service name, topic and filter on local member.
+     *
+     * @param serviceName the service name for which we are registering
+     * @param topic       the event topic for which we are registering
+     * @param filter      the filter for the listened events
+     * @param listener    the event listener
+     * @return the event registration
+     * @throws IllegalArgumentException if the listener or filter is null
+     */
+    private EventRegistration registerListenerInternalLocally(@Nonnull String serviceName, @Nonnull String topic,
```

Comment:
maybe we can call this just `registerListener`?

Reply:
See https://github.com/hazelcast/hazelcast/pull/15860#discussion_r347896293